datamol/match.py

```diff
@@ -0,0 +1,139 @@
+from rdkit.Chem import AllChem
+from typing import Dict, List, Union
+import networkx as nx
+from networkx.algorithms.isomorphism.vf2userfunc import GraphMatcher
+from loguru import logger
+
+
+def mol_to_nx_graph(mol: AllChem.rdchem.Mol) -> nx.Graph:
+    """
+    Convert a molecule to a networkx graph.
+
+    Parameters:
+        mol: Molecule to convert
+    Returns:
+        A networkx graph containing an `atomic_num` attribute on the nodes
+        and a `weight` attribute on the edges, corresponding to
+        single=1, double=2, triple=3, quadruple=4, quintuple=5, aromatic=1.5.
```

Comment:
[unrelated to PR] quadruple and quintuple bonds, are potentially not going to appear in organic chemistry. In your use case, did you have a lot of transition metals ?

Reply:
Changed behaviour to consider any BondType.

Reply:
@maclandrol , saw this function, but in my case, I also needed to match atomic charges since they're not available in `.xyz` file but might be present in the molecule.

Reply:
Are you using [xyz2mol](https://github.com/jensengroup/xyz2mol) first for conversion ?

Reply:
Tried different solutions, including xyz2mol and openbabel. Doesn't work 100% of the time.